### PR TITLE
[WIP][Draft] Initial Koji/Standard Dist Support

### PR DESCRIPTION
**Pull request by ryncsn**
Built and Tested on TS2 & TS3.

Now to build the kernel package, simply run
```
make dist-rpm
```
In repo root.

Or, to get a rebuild-able SRPM, run:

```
make dist-srpm
```

The output log will indicate the built RPM/SRPM location.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/TencentOS-kernel?pullRequest=299) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/TencentOS-kernel?pullRequest=299) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/TencentOS-kernel?pullRequest=299) it.</sub>

File: dist/kabi/Module.kabi_x86_64 [hunk 1]
```diff
@@ -0,0 +1 @@
+../../package/default/Module.kabi_x86_64
```
[the remark this answers is not in the document]
Comment (ryncsn) [reply dist/kabi/Module.kabi_x86_64:1]: It was generated manually, but cherry-picking symbols from symver file. And kABI have to be specified manually. If no kABI candidate was chosen, just leave this file as an empty file, kABI related functionality will be disabled.

File: dist/scripts/check-commits.sh [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+}
+
+# Check if a commit id is valid upstream commit id
+# $1: commit id to check
+is_valid_upstream_commit() {
+	local commit=$1
+	local upstream_repo=(
+		"torvals https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/patch/?id=<>"
+		"stable https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/patch/?id=<>"
+		"tip https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/patch/?id=<>"
```
[the remark this answers is not in the document]
Comment (ryncsn) [reply dist/scripts/check-commits.sh:108]: Right, I forgot to update this boilerplate... This is used to check if the an upstream commit id is valid, it seems some git repo on git.kernel.org shares a large git base repo, so looking for a commit in a different repo actually works, and so did this... But still I will fix it, this is not an expected behavior.

File: package/default/config.default [hunk 3]
```diff
@@ -183,8 +183,8 @@ CONFIG_INITRAMFS_SOURCE=""
 CONFIG_RD_GZIP=y
 # CONFIG_RD_BZIP2 is not set
 CONFIG_RD_LZMA=y
-# CONFIG_RD_XZ is not set
-# CONFIG_RD_LZO is not set
+CONFIG_RD_XZ=y
```
[the remark this answers is not in the document]
Comment (ryncsn) [reply package/default/config.default:186]: The ISO build will compress the initramfs in XZ format, this is required.

File: dist/kabi/Module.kabi_aarch64 [hunk 4]
```diff
@@ -0,0 +1 @@
+../../package/arm/Module.kabi_aarch64
```
[the remark this answers is not in the document]
Comment (ryncsn) [reply dist/kabi/Module.kabi_aarch64:1]: It is possible but maybe a smooth transition is better?

File: dist/scripts/gen-spec.sh [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+%define kernel_relver $KERNEL_RELVER
+EOF
+[ "$KERNEL_VARIANT" ] && cat << EOF
+%define kernel_variant $KERNEL_VARIANT
+EOF
+}
+
+_gen_arch_source() {
+	# Source1000 - Source1199 for kernel config
+	local config_source_num=1000 arch
```
[the remark this answers is not in the document]
Comment (ryncsn) [reply dist/scripts/gen-spec.sh:116]: It for allocation source file id in spec file. Each source file need a unique id, each number stands for a file. We will definitely not have up to 200 kABI files (usually just 1 for each arch, ~10 at most I guess).

File: dist/kabi/Module.kabi_x86_64 (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexshi) [reply dist/kabi/Module.kabi_x86_64:1]: the file Module.kabi... has 14k lines, unlike some selected symbols.... and no arm kabi file...

**Comment by alexshi** [comment]
It's fine as a start point. But what's a normal kabi symbols numbers?  maybe 14k is too big?

**Comment by alexshi** [comment]
btw, please highlight the kernel name changes at somewhere

**Comment by alexshi** [comment]
Reviewed-by: Alex Shi <alexsshi@tencent.com>

**Pull request merged**